### PR TITLE
Small doc corrections

### PR DESCRIPTION
**Pull request by MatKuhr**
This PR ships small updates to the OData generator documentation and corrects a mistake in the typed OData V2 API usage.

### CODE REVIEW

File: docs/java/features/odata/generate-client.mdx [hunk 1]
```diff
@@ -97,94 +97,74 @@ Furthermore, some common IDEs (e.g. IntelliJ, Eclipse) require plugins to recogn
...
+            <phase>generate-sources</phase>
+            <goals>
+                <goal>generate</goal>
+            </goals>
+            <configuration>
+                <inputDirectory>${project.basedir}/edmx</inputDirectory>
+                <outputDirectory>${project.build.directory}/vdm</outputDirectory>
+                <deleteOutputDirectory>true</deleteOutputDirectory>
+                <packageName>com.mycompany.vdm</packageName>
+                <defaultBasePath>odata/v4/</defaultBasePath>
```
Comment (KavithaSiva) [inline docs/java/features/odata/generate-client.mdx:124]:
```suggestion
                <defaultBasePath>odata/v2/</defaultBasePath>
```

**Review by KavithaSiva — COMMENTED** [review]
Thanks for taking care Matthias. Except for one small suggestion from my side, LGTM.

File: docs/java/features/odata/generate-client.mdx (hunk 1 above)
Comment (MatKuhr) [reply docs/java/features/odata/generate-client.mdx:124]: The suggestion was good but on the wrong section ;)

Comment (KavithaSiva) [reply docs/java/features/odata/generate-client.mdx:124]: How do you find the text that corresponds to a particular tab when you have multiple tabs? I find it difficult to find in a markdown file.

**Review by KavithaSiva — APPROVED** [review]

File: docs/java/features/odata/generate-client.mdx (hunk 1 above)
Comment (MatKuhr) [reply docs/java/features/odata/generate-client.mdx:124]: Look for the `<TabItem value="v2">` above the code. Everything between that and `</TabItem>` belongs to the `v2` tab.

Comment (KavithaSiva) [reply docs/java/features/odata/generate-client.mdx:124]: Ah okay, missed that. Thanks for pointing it out.

**Pull request merged**